src/pcrlock/pehash.c

```diff
@@ -216,10 +216,23 @@ int uki_hash(int fd,
                 if (EVP_DigestInit_ex(mdctx, md, NULL) != 1)
                         return log_debug_errno(SYNTHETIC_ERRNO(ENOTRECOVERABLE), "Failed to allocate message digest.");
 
-                r = hash_file(fd, mdctx, section->PointerToRawData, section->VirtualSize);
+                r = hash_file(fd, mdctx, section->PointerToRawData, MIN(section->VirtualSize, section->SizeOfRawData));
                 if (r < 0)
                         return r;
 
+                if (section->SizeOfRawData < section->VirtualSize) {
+                        size_t remaining = section->VirtualSize - section->SizeOfRawData;
+
+                        while (remaining > 0) {
+                                size_t sz = MIN(1024U, remaining);
+
+                                if (EVP_DigestUpdate(mdctx, (char[1024]) {}, sz) != 1)
```

Comment:
would prefer if we'd define a buffer outside the while loop we initialize only once, i.e just do:

```c
uint8_t zeroes[1024] = {};
```

immediately before the `while`.

And then maybe change the MIN() line above to use `sizeof(zeroes)` instead of `1024U` as first parameter.